packages/@aws-cdk/aws-eks/README.md

```diff
@@ -521,6 +522,51 @@ const cluster = new eks.Cluster(this, 'hello-eks', {
 
 The default value is `eks.EndpointAccess.PUBLIC_AND_PRIVATE`. Which means the cluster endpoint is accessible from outside of your VPC, but worker node traffic and `kubectl` commands issued by this library stay within your VPC.
 
+### Alb Controller
+
+Some Kubernetes resources are commonly implemented on AWS with the help of the [ALB Controller](https://kubernetes-sigs.github.io/aws-load-balancer-controller/v2.3/).
+
+From the docs:
+
+> AWS Load Balancer Controller is a controller to help manage Elastic Load Balancers for a Kubernetes cluster.
+>
+> * It satisfies Kubernetes Ingress resources by provisioning Application Load Balancers.
+> * It satisfies Kubernetes Service resources by provisioning Network Load Balancers.
+
+To deploy the controller on your EKS cluster, configure the `albController` property:
+
+```ts
+new eks.Cluster(this, 'HelloEKS', {
+  version: eks.KubernetesVersion.V1_21,
+  albController: {
+    version: eks.AlbControllerVersion.V2_3_0,
+  },
+});
+```
+
+Querying the controller pods should look something like this:
+
+```console
+❯ kubectl get pods -n kube-system
+NAME                                            READY   STATUS    RESTARTS   AGE
+aws-load-balancer-controller-76bd6c7586-d929p   1/1     Running   0          109m
+aws-load-balancer-controller-76bd6c7586-fqxph   1/1     Running   0          109m
+...
+...
+```
+
+Every Kubernetes manifest that utilizes the ALB Controller is effectively dependant on the controller.
+If the controller is deleted before the manifest, it might result in dangling ELB/ALB resources.
+Currently, the EKS construct library does not detect such dependencies, and they should be done explicitly.
+
+For example:
+
+```ts
+declare const cluster: eks.Cluster;
+const manifest = cluster.addManifest('manifest', {/* ... */});
+manifest.node.addDependency(cluster.albController ?? [])
```

Comment:
I actually started of doing this. The problem is that this would create a circular dependency since the ALB itself creates a `ServiceAccount`, which is a manifest. 

We would need to add an awkward `dependsOnALB` option to `KubernetesManifest` which I didn't want to get into now. I'm also not sure adding this to ALL manifests is the right thing...

I think its ok to start of with this, and after we can some bake time maybe we can reduce this friction.